android/sdk/src/main/java/com/taobao/weex/ui/view/listview/adapter/WXRecyclerViewOnScrollListener.java

```diff
@@ -266,46 +276,49 @@ public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
     super.onScrolled(recyclerView, dx, dy);
     RecyclerView.LayoutManager layoutManager = recyclerView.getLayoutManager();
     IOnLoadMoreListener l;
-    if((l = listener.get()) != null){
-      l.onBeforeScroll(dx,dy);
+    if ((l = listener.get()) != null) {
+      l.onBeforeScroll(dx, dy);
     }
 
     //  int lastVisibleItemPosition = -1;
-    if (layoutManagerType == null) {
-      if (layoutManager instanceof LinearLayoutManager) {
-        layoutManagerType = LAYOUT_MANAGER_TYPE.LINEAR;
-      } else if (layoutManager instanceof GridLayoutManager) {
-        layoutManagerType = LAYOUT_MANAGER_TYPE.GRID;
-      } else if (layoutManager instanceof StaggeredGridLayoutManager) {
-        layoutManagerType = LAYOUT_MANAGER_TYPE.STAGGERED_GRID;
-      } else {
-        throw new RuntimeException(
-            "Unsupported LayoutManager used. Valid ones are LinearLayoutManager, GridLayoutManager and StaggeredGridLayoutManager");
-      }
-    }
+    if (layoutManager instanceof LinearLayoutManager) {
+      layoutManagerType = LAYOUT_MANAGER_TYPE.LINEAR;
+      mLastVisibleItemPosition = ((LinearLayoutManager) layoutManager).findLastVisibleItemPosition();
+      listener.get().notifyAppearStateChange(((LinearLayoutManager) layoutManager).findFirstVisibleItemPosition()
```

Comment:
1. listener can be null
2.`listener.get().notifyAppearStateChange(` is same in other branch, put this call out of 'if', code would be more readable.